src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -819,32 +792,233 @@ bool AutomationInstrumentClipView::renderSidebar(uint32_t whichRows, uint8_t ima
 	return instrumentClipView.renderSidebar(whichRows, image, occupancyMask);
 }
 
-//adjust the LED meters
-void AutomationInstrumentClipView::displayAutomation() {
+//render's what is displayed on OLED or 7SEG screens when in Automation View
```

Comment:
new function created to render information on OLED and 7Seg displays

On Automation Overview:
- on OLED it renders "Automation Overview" (or "Can't Automate CV" if you're on a CV clip)
- on 7Seg it renders AUTO (or CANT if you're on a CV clip)

On Automation Editor:
- on OLED it renders Parameter Name, Automation Status and Parameter Value (for selected Pad or the current value for the Parameter for the last selected Mod Position)
- on 7SEG it renders Parameter name if no pad is selected or mod encoder is turned. If selecting pad it displays the pads value for as long as you hold the pad. if turning mod encoder, it displays value while turning mod encoder. after value displaying is finished, it displays scrolling parameter name again.

This function replaces the two functions that were previously called:
- DisplayParameterValue
- DisplayParameterName

Reply:
Can you make this explanation into a doc comment?

Reply:
Also split to a 7seg and oled function for readability instead of the giant if/else please